tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/executable/cases/project/NewEspressifIDFProjectSBOMTest.java

```diff
@@ -0,0 +1,216 @@
+	{
+		Fixture.givenNewEspressifIDFProjectIsSelected("EspressIf", "Espressif IDF Project");
+		Fixture.givenProjectNameIs("NewSbomProjectFourthTest");
+		Fixture.whenNewProjectIsSelected();
+		Fixture.whenProjectIsBuiltUsingContextMenu();
+		Fixture.whenOpenSBOMtool();
+		Fixture.thenCheckPathValidation(Fixture.bot);
+	}
+
+	private static class Fixture
+	{
+		private static SWTWorkbenchBot bot;
+		private static String category;
+		private static String subCategory;
+		private static String projectName;
+
+		private static void loadEnv() throws Exception
+		{
+			bot = WorkBenchSWTBot.getBot();
+			EnvSetupOperations.setupEspressifEnv(bot);
+			bot.sleep(1000);
+		}
+
+		private static void givenNewEspressifIDFProjectIsSelected(String category, String subCategory)
+		{
+			Fixture.category = category;
+			Fixture.subCategory = subCategory;
+		}
+
+		private static void givenProjectNameIs(String projectName)
+		{
+			Fixture.projectName = projectName;
+		}
+
+		private static void whenNewProjectIsSelected() throws Exception
+		{
+			ProjectTestOperations.setupProject(projectName, category, subCategory, bot);
+		}
+
+		private static void whenProjectIsBuiltUsingContextMenu() throws IOException
+		{
+			ProjectTestOperations.buildProjectUsingContextMenu(projectName, bot);
+			ProjectTestOperations.waitForProjectBuild(bot);
+			TestWidgetWaitUtility.waitForOperationsInProgressToFinishAsync(bot);
+		}
+
+		private static void cleanTestEnv()
+		{
+			TestWidgetWaitUtility.waitForOperationsInProgressToFinishAsync(bot);
+			ProjectTestOperations.closeAllProjects(bot);
+			ProjectTestOperations.deleteAllProjects(bot);
+		}
+
+		private static void whenOpenSBOMtool() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "SBOM Tool");
+		}
+
+		private static void thenRunOkbuttonDisabled(SWTWorkbenchBot bot) throws IOException
+		{
+			SWTBotButton okButton = bot.button("OK");
+			assertTrue(!okButton.isEnabled());
+			bot.checkBox("Redirect output to the file").click();
+			assertTrue(!okButton.isEnabled());
+			bot.checkBox("Redirect output to the file").click();
+			assertTrue(!okButton.isEnabled());
+			bot.button("Cancel").click();
+		}
+
+		private static void whenRunSBOMtool() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "SBOM Tool");
+			bot.button("OK").click();
+			ProjectTestOperations
+					.joinJobByName(com.espressif.idf.ui.update.Messages.SbomCommandDialog_EspIdfSbomJobName);
+		}
+
+		private static void thenCheckResultInConsole() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console", "SPDXVersion");
+		}
+
+		private static void thenVerifyConsole() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console",
+					"The output was redirected to the file:");
+		}
+
+		private static void whenRunSBOMtoolUsingContextMenuRedirectOutputToFile() throws IOException
```

Comment:
This method should only be used to redirect output to file you already have a method for opening sbom